recording-daemon/metafile.c

```diff
@@ -149,11 +149,33 @@ static void meta_ptime(metafile_t *mf, unsigned long mnum, int ptime)
 	mf->media_ptimes[mnum] = ptime;
 }
 
+static char *get_output_path(metafile_t *mf) {
+	char *res = NULL;
+	str all_meta;
+	str_init(&all_meta, mf->metadata);
+	while (all_meta.len > 1) {
+		str token;
+		if (str_token_sep(&token, &all_meta, '|'))
+			break;
+		str key;
+		if (str_token(&key, &token, ':')) {
+			// key:value separator not found, skip
+			continue;
+		}
+		if (strncmp(key.s, "output-dir", 10) >= 0) {
+			ilog(LOG_INFO, "Metadata output-dir found token=%s", token.s);
+			res = token.s;
+			break;
+		}
+	}
+	return res;
+}
```

Comment:
This duplicates a function in db.c. Are you able to refactor this in such a way that the metadata is parsed only once, and then used for both purposes (e.g. parse into a hash table or list and store into `mf` or something like that)?

Another option would be not to (ab)use the metadata for this purpose at all but instead pass the output path as a separate section in the metafile (e.g. similar to the `metadata` string itself, but an extra string just for that purpose).